src/AccountSimulator/src/Smart.FA.Catalog.AccountSimulator/AccountData.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Text.Json;
+
+namespace Smart.FA.Catalog.AccountSimulator;
+
+public class AccountData
+{
+    public string? FirstName { get; set; }
+    public string? LastName { get; set; }
+    public string? Email { get; set; }
+}
+
+public static class AccountDataFactory
+{
+    private static AccountData CreateMock(string firstName, string lastName, string email)
+    {
+        return new AccountData { FirstName = firstName, LastName = lastName, Email = email };
+    }
+
+    public static string Serialize(this AccountData accountData)
+    {
+        var serializedCustomData = JsonSerializer.Serialize(accountData);
+        return System.Web.HttpUtility.UrlDecode(serializedCustomData);
+    }
+
+    public static AccountData? Deserialize(string serializedCustomData)
+    {
+        if (string.IsNullOrEmpty(serializedCustomData))
+        {
+            return new AccountData();
+        }
+
+        return JsonSerializer.Deserialize<AccountData?>(serializedCustomData);
+    }
```

Comment:
I noticed that you have a serializedAccountData below so I guess it should be named the same here.
```suggestion
    public static string Serialize(this AccountData accountData)
    {
        var serializedAccountData = JsonSerializer.Serialize(accountData);
        return System.Web.HttpUtility.UrlDecode(serializedAccountData );
    }

    public static AccountData? Deserialize(string serializedAccountData)
    {
        if (string.IsNullOrEmpty(serializedAccountData))
        {
            return new AccountData();
        }

        return JsonSerializer.Deserialize<AccountData?>(serializedAccountData );
    }
```